public/services/OpenSearchService.ts

```diff
@@ -56,4 +57,28 @@ export default class OpenSearchService {
 
     return Promise.resolve(indexPatterns);
   };
+
+  getFieldsForWildcard(options: GetFieldsOptions = {}) {
```

Comment:
@AWSHurneyt can we reuse similar client side service method from Dashboards Core?

Reply:
Yes, we can make use of that method if it's in core. The alerting frontend depends on dashboards core, so it should always be available. I would recommend implementing a test within the alerting frontend plugin that confirms it's working as expected though just in case changes are made to the method in core.

Reply:
@AWSHurneyt it is in the core repository, but in [plugins](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts#L233). As far as I found, it's possible to import something from [src/core/public](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/public/index.ts) with imports like this:
```ts
import { CoreStart, AppMountParameters } from 'opensearch-dashboards/public';
```
I cannot find any example usage of something out of core/public. Let me know if you have some example in alerting plugin that you mentioned.